pkg/providers/amifamily/ami.go

```diff
@@ -160,8 +160,41 @@ func (p *Provider) getDefaultAMIFromSSM(ctx context.Context, nodeTemplate *v1alp
 		if err != nil {
 			return nil, err
 		}
-		amis = append(amis, AMI{Name: ssmOutput.Name, AmiID: amiID, Requirements: ssmOutput.Requirements})
+		amis = append(amis, AMI{AmiID: amiID, Requirements: ssmOutput.Requirements})
 	}
+
+	amis, err = p.findAMINames(ctx, amis)
+	if err != nil {
+		return nil, err
+	}
+
+	return amis, nil
+}
+
+func (p *Provider) findAMINames(ctx context.Context, amis []AMI) ([]AMI, error) {
```

Comment:
It's not immediately clear to me what this function does so can we add a function comment to say what this function does?